iceoryx_utils/include/iceoryx_utils/internal/cxx/static_storage.inl

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_CXX_STATIC_STORAGE_INL
+#define IOX_UTILS_CXX_STATIC_STORAGE_INL
+
+#include "iceoryx_utils/internal/cxx/static_storage.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+template <uint64_t Capacity, uint64_t Align>
+constexpr uint64_t static_storage<Capacity, Align>::align_delta(uint64_t align, uint64_t alignTarget)
```

Comment:
The arguments could be const ref.

Reply:
Also, why snake case here ? Does this class reimplement the STL ?

Reply:
Nope, but it is low level and I like it in this case. I really dislike Upper Camel Case for those kinds of constructions (similar with e.g. relative_ptr, which are supposed to mimic shared_ptr syntax).

Reply:
The args are primitive types, same size as pointers on a 64 bit system. No point passing them by ref IMO.